serverless/serverless
# Fix #1168: add ability to specify request pass through behavior for …

***Implementing Issue:*** #1168 

## What did you implement:
Added ability to specify a pass-through behavior for API Gateway requests.  It requires that you select one of the valid methods (NEVER, WHEN_NO_MATCH, WHEN_NO_TEMPLATES), and uses NEVER as the default value if none was specified.

## How did you implement it:
I updated the apiGateway events method processing to include a check for the "passThrough" configuration.  If none specified, it will use the default value.  If an invalid value is passed, an error will be thrown.

## How can we verify it:
Update your function config under events.http.request to include a "passThrough" value.

```yml
functions:
  create:
    handler: posts.create
    events:
      - http:
          method: get
          path: whatever
          request:
            passThrough: WHEN_NO_TEMPLATES
```

You can verify that the resulting CloudFormation template under Integration -> PassthroughBehavior has the correct value.
 
Once deployed, you can confirm in the API Gateway dashboard under that method -> Integration Request -> Body Mapping Templates.

## Todos:

- [x] Write docs

…API Gateway
